src/providers/discordRpcProvider.js

```diff
@@ -41,70 +42,148 @@ async function setActivity(info) {
+                        numberOfBars
+                )
+
+                let status = ''
+                let replacer = ''
+
+                for (let index = 0; index < numberOfBars; index++) {
+                    if (index == progress) replacer = progressBarD.currentTime
+                    else if (index < progress) replacer = progressBarD.fill
+                    else replacer = progressBarD.base
+                    status += replacer
+                }
+
+                activity.state = status
+            }
+
+            if (discordSettings.time) {
+                if (info.player.isPaused) {
+                    delete activity.startTimestamp
+                    delete activity.endTimestamp
+                } else {
+                    activity.startTimestamp =
+                        now + info.player.seekbarCurrentPosition * 1000
+                    activity.endTimestamp =
+                        now +
+                        (info.track.duration -
+                            info.player.seekbarCurrentPosition) *
+                            1000
+                }
+            }
+
+            // activity.largeImageKey = 'ytm_logo_512'
+            activity.largeImageKey = info.track.cover
+            activity.smallImageKey = info.player.isPaused
+                ? 'discordrpc-pause'
+                : 'discordrpc-play'
+            activity.largeImageText = 'YouTube Music'
+            activity.smallImageText = info.player.isPaused
+                ? 'Paused'
+                : 'Playing'
+            activity.instance = false
+            if (discordSettings.details) {
+                activity.buttons = [
+                    {
+                        label: 'Play on YouTube Music',
+                        url:
+                            'https://music.youtube.com/watch?v=' +
+                            info.track.id,
                     },
-                    instance: activity.instance,
-                    buttons: activity.buttons,
-                },
-            })
+                ]
+            }
+        }
+
+        discordActivity = {
+            state: activity.state,
+            details: activity.details,
+            timestamps: {
+                start: activity.startTimestamp,
+                end: activity.endTimestamp,
+            },
+            assets: {
+                large_image: activity.largeImageKey,
+                large_text: activity.largeImageText,
+                small_image: activity.smallImageKey,
+                small_text: activity.smallImageText,
+            },
+            instance: activity.instance,
+            buttons: activity.buttons,
         }
     }
 }
 
+setInterval(async () => {
+    if (
+        (!discordSettings.hideIdle
+    )
+        await client.clearActivity()
+    else
+        await client?.request('SET_ACTIVITY', {
+            pid: process.pid,
+            activity: discordActivity,
+        })
+}, 5000)
```

Comment:
This should be moved to a new place so that it's not applied to ALL users, remember that this file will still get ran if the user has Discord RPC disabled.

Additionally I feel that there could be some clean up regarding setting the Discord Activity to null when hideIdle is off & we've already cleared the activity to not keep running it.